### ISSUE TITLE
Why memory exceeded

### ISSUE THREAD

**Issue by CashCode** [php-curl-class/php-curl-class#321]
Hey.
I run simple script like this:
```
for($i=1; $i < ceil($totalUrls/$threads); $i++) {
		$multi_curl = null;
		$multi_curl = new MultiCurl();
		$multi_curl->setOptArray(array(CURLOPT_SSL_VERIFYHOST => false, CURLOPT_SSL_VERIFYPEER => false));
		$multi_curl->success(function($instance) {
			preg_match_all('#something#', $instance->response, $m);
			$ids = array_unique($m[1]);
			file_put_contents('./data.txt', implode("\r\n", $ids)."\r\n", FILE_APPEND);
			echo $instance->url . '" success. Parsed:' . count($ids) . "\n";
		});
		$multi_curl->error(function($instance) {
		    echo 'call to "' . $instance->url . '" was unsuccessful.' . "\n";
		    echo 'error code: ' . $instance->errorCode . "\n";
		    echo 'error message: ' . $instance->errorMessage . "\n";
		});
	foreach(range($i*$threads+1, ($i+1)*$threads) as $pageNum) 
	{
		$multi_curl->addGet('http://example/'.$pageNum);
	}
	$multi_curl->start();
	$multi_curl->close();
	echo "\n";
}
```
And php-cli starts eating memory more and more until it's over.
![y7ufthi](https://cloud.githubusercontent.com/assets/8258294/14228356/27fe1dd4-f91b-11e5-9db2-5bf721144aca.png)

Why it's so?

**Comment by thebeline** [comment]
Because it does not garbage collect, ever, due to a recursive dependency.  See #310 WEEEE

**Comment by thebeline** [comment]
ALSO, you might be working on a different branch or something, as there is no `setOptArray` currently.  I implemented something like that with #271, but, for some strange reason, I can get a dentist apt before getting a PR accepted.

@zachborboa :-)

**Comment by CashCode** [comment]
setOptArray  is my own fucntion, it just call setOpt for array. But I didn't download anything. I can't understand this fix, what should I change? Explain me more widely, if possible ) thanks

**Comment by thebeline** [comment]
You won't be "downloading a fix," not unless you like juggling detached heads, anyway.  Best bet is to wait for it to be merged in to `master` and pull it at that point.  Either that or you can manually patch it using the code from the linked pull request.

It may be a simple merge, as well.  Go ahead and save your changes to your local branch, and try to pull my branch in, it might "just work."

**Comment by zachborboa** [comment]
Hi @CashCode the memory issues should now be fixed (3789e2077b0c6fe3acc828c1235cf01726b3e1fe). Try the latest 6.0.0 release. Thanks
